_posts/2017-02-14-planning-for-7.0.md

```diff
@@ -0,0 +1,194 @@
+
+To simplify things, we've decided to standardize on requiring presets be functions that return a Babel config object (rather than just an object).
+
+Before (invalid in 7.0)
+
+```js
+export default {
+  plugins: ['plugin-a']
+};
+```
+
+After
+
+```js
+export default function() { // or module.exports
+  return {
+    plugins: ['plugin-a']
+  };
+}
+```
+
+## Meta
+
+### [#5218](https://github.com/babel/babel/pull/5218) Remove `babel-runtime` from Babel dependencies
+
+Babel itself doesn't have that many external dependencies, but in 6.x [each package has a dependency on `babel-runtime`](https://github.com/babel/babel/blob/958f72ddc28e2f5d02adf44eadd2b1265dd0fa4d/packages/babel-plugin-transform-es2015-arrow-functions/package.json#L12) so that built-ins like Symbol, Map, Set, and others are available without needing a polyfill. By changing the minimum supported version of Node to v4 (where those built-ins are supported natively), we can drop the dependency entirely.
+
+> This is an issue on npm 2 (we didn't recommended it's use with Babel 6) and older yarn, but not npm 3 due to its deduping behavior.
+
+With [Create React App](https://github.com/facebookincubator/create-react-app) the size of the node_modules folder changed drastically when babel-runtime was hoisted.
+
+- `node_modules` for npm 3: ~120MB
+- `node_modules` for Yarn (<`0.21.0`): ~518MB
+- `node_modules` for Yarn (<`0.21.0`) with hoisted `babel-runtime`: ~157MB
+- `node_modules` for Yarn + [PR #2676](https://github.com/yarnpkg/yarn/pull/2676): ~149MB ([tweet](https://twitter.com/bestander_nz/status/833696202436784128))
+
+So although this issue will be fixed "upstream" by using npm 3/latest Yarn, we can do our part by simply dropping our own dependency on `babel-runtime`.
+
+### [#5224](https://github.com/babel/babel/pull/5224) Independent Publishing of Packages
+
+> I mention this in [The State of Babel](http://babeljs.io/blog/2016/12/07/the-state-of-babel#the-future-versioning).
+> https://github.com/babel/babylon/issues/275
+
+You might remember that after Babel 6, Babel became a set of npm packages with its own ecosystem of custom presets and plugins.
+
+However since then, we've always used a "fixed/synchronized" versioning system (so that no package is on v7.0 or above). When we do a new release such as `v6.23.0` only packages that have updated code in the source are published with the new version while the rest of the packages are left as is. This mostly works in practice because we use `^` in our packages.
+
+Unfortunately this kind of system requires that a major version be released for all packages once a single package needs it. This either means we make a lot small breaking changes (unnecessary) or we batch lots of breaking changes into a single release. This is the same issue for a single package/repo but we have a way out: since Babel is a monorepo of npm packages, we can actually publish breaking changes in individual packages if necessary.
```

Comment:
Instead of "we can actually publish breaking changes in individual packages if necessary." just go with the summary from the "This just means" section and lead with making major version updates. If you lead with breaking changes folks will panic before getting to the next paragraph.